tests/frontier/opcodes/test_call_and_callcode_gas_calculation.py

```diff
@@ -0,0 +1,159 @@
+"""
+abstract: Tests the nested CALL/CALLCODE opcode gas consumption with a positive value transfer.
+
+    This test is designed to investigate an issue identified in EthereumJS, as reported in:
+    https://github.com/ethereumjs/ethereumjs-monorepo/issues/3194.
+
+    The issue pertains to the incorrect gas calculation for CALL/CALLCODE operations with a
+    positive value transfer, due to the pre-addition of the gas stipend (2300). This bug was
```

Comment:
```suggestion
    positive value transfer, due to the pre-addition of the gas stipend (2300) to the currently available gas instead of adding it to the new call frame. This bug was
```